### PR TITLE
Small Update to troubleshooting document

### PR DESCRIPTION
**Pull request by jasonwilliams14**
### Proposed changes
This is a small update to our troubleshooting guide adding additional information. This will be an ongoing process as we continue to upgrade and provide additional documentation for NGINX Ingress Controller.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] I have checked that all unit tests pass after adding my changes
- [ x] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 1]
```diff
@@ -47,102 +47,144 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
 $ kubectl logs <nginx-ingress-pod> -n nginx-ingress
 ```
 
-Controlling the verbosity and format:
-* To control the verbosity of the Ingress Controller software logs (from 1 to 4), use the `-v` [command-line argument](/nginx-ingress-controller/configuration/global-configuration/command-line-arguments). For example, with `-v=3` you will get more information and the content of any new or updated configuration file will be printed in the logs.
-* To control the verbosity and the format of the NGINX logs, configure the corresponding [ConfigMap keys](/nginx-ingress-controller/configuration/global-configuration/configmap-resource).
+### Enabling debuging for NGINX Ingress Controller
 
-### Checking the Events of an Ingress Resource
+If you need to do additional troubleshooting for NGINX Ingress controller, there are few additional settings you can configure, to add more verbose logging.
```
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:52]:
```suggestion
If you need to do additional troubleshooting for the Ingress Controller, there are a few additional settings you can configure, to add more verbose logging.
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 2]
```diff
@@ -47,102 +47,144 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
-```
-$ kubectl describe pol webapp-policy
-. . .
-Events:
-  Type    Reason          Age   From                      Message
-  ----    ------          ----  ----                      -------
-  Normal  AddedOrUpdated  11s   nginx-ingress-controller  Policy default/webapp-policy was added or updated
-```
-Note that in the events section, we have a `Normal` event with the `AddedOrUpdated` reason, which informs us that the policy was successfully accepted.
+There are two settings that need to be set to enable more debug/verbose logging for NGINX Ingress controller.
```
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:54]:
```suggestion
There are two settings that need to be set to enable more debug/verbose logging for the Ingress Controller.
```
Isn't this mostly a repetition of the sentence above?

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 3]
```diff
@@ -47,102 +47,144 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
 
-However, the fact that a policy was accepted doesn't guarantee that the NGINX configuration was successfully applied. To confirm that, check the events of the VirtualServer and VirtualServerRoute resources that reference that policy.
+1. command line arguments
+2. configmap settings
+
+This document will cover how to enable both.
+
+
+You can use NGINX Ingress controller command line arguments. This is a great way to increase debug log levels for both Ingress as well as NGINX.
```
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:62]:
```suggestion
You can use the Ingress Controller command line arguments. This is a great way to increase debug log levels for both the Ingress Controller as well as NGINX.
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 4]
```diff
@@ -47,102 +47,144 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
+When using `manifest` for deployment:
+
+1.) Use the command line argument, `- -nginx-debug` in your deployment/daemonset
+2.) If you want to increase the Ingress `Controller` process lets, set the command line argument to `v=3`, which collects more information on Ingress
```
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:66]:
```suggestion
1.) Use the command line argument `- -nginx-debug` in your deployment or daemonset
2.) If you want to increase the verbosity of the Ingress Controller process, set the command line argument to `v=3`
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 5]
```diff
@@ -47,102 +47,144 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
+
+ ```yaml
+ -v=3
+ ```
+
+Here is a small snippet of setting these a command line arguments in the `ARGS` section of a deployment:
```
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:72]:
```suggestion
Here is a small snippet of setting these command line arguments in the `args` section of a deployment:
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 6]
```diff
@@ -47,102 +47,144 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
+
+```yaml
+args:
+  - -nginx-configmaps=$(POD_NAMESPACE)/nginx-config
+  - -enable-cert-manager
+  - -nginx-debug
+  - -v=3
+```
+
+You can configure `error-log-level` in the NGINX Ingress controller `configMap`:
```
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:82]:
```suggestion
You can configure `error-log-level` in the Ingress Controller `configMap`:
```

**Review by lucacome — COMMENTED** [review]
We want to remove the other sections about checking the config, etc?

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md (hunk 6 above)
Comment (ADubhlaoich) [reply docs/content/troubleshooting/troubleshoot-ingress-controller.md:82]:
```suggestion
You can configure `error-log-level` in the NGINX Ingress Controller `configMap`:
```

The name should be capitalised, but we actually keep the 'NGINX' prefix to product names at all times.

File: docs/content/troubleshooting/troubleshoot-configmap-policy.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+Checking the Events of a Policy Resource
+After you create or update a Policy resource, you can use kubectl describe to check whether or not the Ingress Controller accepted the Policy:
+
+$ kubectl describe pol webapp-policy
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-configmap-policy.md:4]:
```suggestion
```kubectl describe pol webapp-policy```
```

Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-configmap-policy.md:4]:
```suggestion
`kubectl describe pol webapp-policy`
```

File: docs/content/troubleshooting/troubleshoot-configmap-policy.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  ----    ------          ----  ----                      -------
+  Normal  AddedOrUpdated  11s   nginx-ingress-controller  Policy default/webapp-policy was added or updated
+Note that in the events section, we have a Normal event with the AddedOrUpdated reason, which informs us that the policy was successfully accepted.
+
+However, the fact that a policy was accepted doesn’t guarantee that the NGINX configuration was successfully applied. To confirm that, check the events of the VirtualServer and VirtualServerRoute resources that reference that policy.
+
+Checking the Events of the ConfigMap Resource
+After you update the ConfigMap resource, you can immediately check if the configuration was successfully applied by NGINX:
+
+$ kubectl describe configmap nginx-config -n nginx-ingress
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-configmap-policy.md:17]:
```suggestion
`kubectl describe configmap nginx-config -n nginx-ingress`
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 9]
```diff
@@ -47,102 +47,160 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
 $ kubectl logs <nginx-ingress-pod> -n nginx-ingress
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:47]:
```suggestion
`kubectl logs <nginx-ingress-pod> -n nginx-ingress`
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 10]
```diff
@@ -47,102 +47,160 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
 ```
 
-Controlling the verbosity and format:
-* To control the verbosity of the Ingress Controller software logs (from 1 to 4), use the `-v` [command-line argument](/nginx-ingress-controller/configuration/global-configuration/command-line-arguments). For example, with `-v=3` you will get more information and the content of any new or updated configuration file will be printed in the logs.
-* To control the verbosity and the format of the NGINX logs, configure the corresponding [ConfigMap keys](/nginx-ingress-controller/configuration/global-configuration/configmap-resource).
+### Enabling debuging for NGINX Ingress Controller
 
-### Checking the Events of an Ingress Resource
+If you need to do additional troubleshooting for the Ingress Controller, there are a few additional settings you can configure, to add more verbose logging.
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:52]:
```suggestion
If you need to do additional troubleshooting for the NGINX Ingress Controller, there are a few additional settings for more verbose logging.
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 11]
```diff
@@ -47,102 +47,160 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
-  Type    Reason   Age                From                      Message
-  ----    ------   ----               ----                      -------
-  Normal  Updated  11s (x2 over 26m)  nginx-ingress-controller  Configuration from nginx-ingress/nginx-config was updated
-```
-Note that in the events section, we have a `Normal` event with the `Updated` reason, which informs us that the configuration was successfully applied.
-
-### Checking the Generated Config
-
-For each Ingress/VirtualServer resource, the Ingress Controller generates a corresponding NGINX configuration file in the `/etc/nginx/conf.d` folder. Additionally, the Ingress Controller generates the main configuration file `/etc/nginx/nginx.conf`, which includes all the configurations files from `/etc/nginx/conf.d`. The config of a VirtualServerRoute resource is located in the configuration file of the VirtualServer that references the resource.
+1. command line arguments
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:56]:
```suggestion
1. Command Line Arguments
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 12]
```diff
@@ -47,102 +47,160 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
+2. configmap settings
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:57]:
```suggestion
2. Configmap Settings
```

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 13]
```diff
@@ -47,102 +47,160 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
+
+This document will cover how to enable both.
+
+
+You can use the Ingress Controller command line arguments. This is a great way to increase debug log levels for both the Ingress Controller as well as NGINX.
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:62]:
```suggestion
Using the command line arguments can be used to increase debug log levels for both the NGINX Ingress Controller as well as NGINX itself.
```

Comment (ADubhlaoich) [reply docs/content/troubleshooting/troubleshoot-ingress-controller.md:62]: Rephrased this slightly to be less "familiar" in tone.

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md [hunk 14]
```diff
@@ -47,102 +47,160 @@ To check the Ingress Controller logs -- both of the Ingress Controller software
...
 $ kubectl exec <nginx-ingress-pod> -n nginx-ingress -- cat /etc/nginx/nginx.conf
 ```
 
-Similarly, you can view the content of any generated configuration file in the `/etc/nginx/conf.d` folder.
+Similarly, you can view the content of any generated configuration file in the /etc/nginx/conf.d folder.
 
 You can also print all NGINX configuration files together:
+
 ```
 $ kubectl exec <nginx-ingress-pod> -n nginx-ingress -- nginx -T
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress-controller.md:198]:
```suggestion
`kubectl exec <nginx-ingress-pod> -n nginx-ingress -- nginx -T`
```

File: docs/content/troubleshooting/troubleshoot-ingress.md [hunk 15]
```diff
@@ -0,0 +1,12 @@
+Checking the Events of an Ingress Resource
+After you create or update an Ingress resource, you can immediately check if the NGINX configuration for that Ingress resource was successfully applied by NGINX:
+
+$ kubectl describe ing cafe-ingress
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-ingress.md:4]:
```suggestion
`kubectl describe ing cafe-ingress`
```

Comment (ADubhlaoich) [reply docs/content/troubleshooting/troubleshoot-ingress.md:4]: @jasonwilliams14 I'm making some small edit suggestions for formatting as I go along; this looks invalid. Should the 'ing' be there?

File: docs/content/troubleshooting/troubleshoot-virtualserver-virtualserverroute.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
+Checking the Events of a VirtualServer and VirtualServerRoute Resources
+After you create or update a VirtualServer resource, you can immediately check if the NGINX configuration for that resource was successfully applied by NGINX:
+
+$ kubectl describe vs cafe
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-virtualserver-virtualserverroute.md:4]:
```suggestion
`kubectl describe vs cafe`
```

File: docs/content/troubleshooting/troubleshoot-virtualserver-virtualserverroute.md [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+. . .
+Events:
+  Type    Reason          Age   From                      Message
+  ----    ------          ----  ----                      -------
+  Normal  AddedOrUpdated  16s   nginx-ingress-controller  Configuration for default/cafe was added or updated
+Note that in the events section, we have a Normal event with the AddedOrUpdated reason, which informs us that the configuration was successfully applied.
+
+Checking the events of a VirtualServerRoute is similar:
+
+$ kubectl describe vsr coffee
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-virtualserver-virtualserverroute.md:14]:
```suggestion
`kubectl describe vsr coffee`
```

File: docs/content/troubleshooting/troubleshoot-configmap-policy.md [hunk 18]
```diff
@@ -0,0 +1,26 @@
+Checking the Events of a Policy Resource
+After you create or update a Policy resource, you can use kubectl describe to check whether or not the Ingress Controller accepted the Policy:
```
Comment (ADubhlaoich) [inline docs/content/troubleshooting/troubleshoot-configmap-policy.md:2]:
```suggestion
After you create or update a Policy resource, you can use `kubectl describe` to check whether or not the Ingress Controller accepted the Policy:
```

File: docs/content/troubleshooting/troubleshoot-configmap-policy.md (hunk 7 above)
Comment (lucacome) [reply docs/content/troubleshooting/troubleshoot-configmap-policy.md:4]: we have `$` in all the commands in the docs...I'm down for removing it everywhere but probably we shouldn't have a mix of both right now?

File: docs/content/troubleshooting/troubleshoot-ingress-controller.md (hunk 6 above)
Comment (lucacome) [reply docs/content/troubleshooting/troubleshoot-ingress-controller.md:82]: this is not consistent in the docs and we actually have more without `NGINX`...do we have to update them all to add it?

File: docs/content/troubleshooting/troubleshoot-configmap-policy.md (hunk 7 above)
Comment (jasonwilliams14) [reply docs/content/troubleshooting/troubleshoot-configmap-policy.md:4]: yea, I agree. Maybe remove the `$` so folks could copy paste without pulling that character in?

File: docs/content/troubleshooting/troubleshoot-ingress.md (hunk 15 above)
Comment (jasonwilliams14) [reply docs/content/troubleshooting/troubleshoot-ingress.md:4]: @ADubhlaoich oo, yea...the `ing` is short for `ingress` in the kubernetes world. :)

**Review by jjngx — APPROVED** [review]
👍🏻 LGTM, but since new updates arrived to the original PR, it may be a good idea to ask @ADubhlaoich to re-check this PR.

File: docs/content/troubleshooting/troubleshoot-transportserver.md
Comment (lucacome) [inline docs/content/troubleshooting/troubleshoot-transportserver.md:1]: I think something was supposed to go in this file 🙂

**Review by ADubhlaoich — CHANGES_REQUESTED** [review]
I'm currently working on a commit to add a lot of missing frontmatter from the newest pages, as well as rename the files - we can explicitly hard-code what the URL slugs for a page will look like, but it's better to have clean paths in the first place.

There's also a lot of product naming issues I am addressing at the same time.

**Review by ADubhlaoich — APPROVED** [review]
I've made the majority of changes I thought were necessary, but there's still additional work needed yet, as well as a review from Jason - shifting to approved, but keeping the PR as a draft.

**Pull request merged**